crates/nu-parser/src/errors.rs

```diff
@@ -109,6 +109,13 @@ pub enum ParseError {
     )]
     CantAliasKeyword(String, #[label("not supported in alias")] Span),
 
+    #[error("Can't create alias to expression.")]
+    #[diagnostic(
+        code(nu::parser::cant_alias_expression),
+        help("Only command calls can be aliased.")
+    )]
+    CantAliasExpression(String, #[label("aliasing {0} is not supported")] Span),
```

Comment:
Maybe worth suggesting using backticks to quote a complex command name?

Reply:
It's not about command name. It's about trying to alias some other expression than a call, like a string, range, etc.

Reply:
Yeah but imagine you try to alias an external command with space in its absolute path like on windows, and the first thing you would possibly try is the string syntax using `"`

Reply:
I think that one you won't be able to detect easily. Imagine a call `foo bar.txt`, you can't know whether you're supposed to call `foo` command with `bar.txt` argument, or a full `foo bar.txt` file. It parses as an external and will skip the math expression test as well, so I can't detect it.

Reply:
I'm gonna merge this because it's blocking my other PR, but we can discuss and improve this further.